exporter/containerimage/writer.go

```diff
@@ -225,6 +234,7 @@ func (ic *ImageWriter) Commit(ctx context.Context, inp *exporter.Source, session
 
 	labels := map[string]string{}
 
+	var descriptors = make(map[string][]*ocispecs.Descriptor)
```

Comment:
This should just be an array of descriptors in the original order without the map. Platform info is already inside the descriptor. Index descriptor is already available in the current result iirc.

Reply:
THat's what I had originally. I can just drop the top commit. The only issue is it needs a subkey because of the way it is unmarshalling it

https://github.com/moby/buildkit/blob/d09703ed9f28d0f400ee65033ee4ad0c2903147a/cmd/buildctl/build.go#L440